ledger-service/http-json/src/main/scala/com/digitalasset/http/ContractsService.scala

```diff
@@ -306,6 +311,44 @@ class ContractsService(
     }
   }
 
+  def refreshCache(
+      jwt: Jwt,
+      jwtPayload: JwtPayload,
+      refreshCacheRequest: RefreshCacheRequest,
+  )(implicit
+      lc: LoggingContextOf[InstanceUUID with RequestID],
+      metrics: HttpJsonApiMetrics,
+  ): Future[SearchResult[Error \/ domain.RefreshCacheResult]] = {
+    val ledgerId = toLedgerId(jwtPayload.ledgerId)
+    val optOffsetToUpdate = refreshCacheRequest.offset
+    getTermination(jwt, ledgerId)(lc).map { optLedgerEnd =>
+      optLedgerEnd.cata(
+        { ledgerEnd =>
+          daoAndFetch.cata(
+            { case (dao, fetchService) =>
+              val response: Source[Error \/ domain.RefreshCacheResult, NotUsed] = {
+                val futureValue =
+                  dao
+                    .transact(
+                      fetchService.fetchAndRefreshCache(jwt, ledgerId, ledgerEnd, optOffsetToUpdate)
+                    )
+                    .unsafeToFuture()
+                Source
+                  .future(futureValue)
+                  .mapConcat(identity)
+                  .map(x => \/.right(domain.RefreshCacheResult(x.toOption)))
```

Comment:
I think this should return the single offset value that we passed into `fetchAndRefreshCache`, i.e. the offset that we ended up refreshing to.